src/NetworkInterface.cpp

```diff
@@ -9870,3 +9872,55 @@ int NetworkInterface::walkActiveHosts(lua_State* vm,
     return(m.info.size());
   }
 }
+
+/* *************************************** */
+
+static bool active_localhosts_walker(GenericHashEntry *h, void *user_data, bool *matched) {
+  Host *host = (Host*)h;
+  localhost_walker *m = (localhost_walker*)user_data;
+  bool isLocal;
+
+  if(host == NULL)
+    return(false);
+
+  isLocal = host->isLocalHost() || host->isSystemHost();
+
+  if(isLocal) {
+  
+    if ((host->getNumPktsSent() == 0) and (host->isBroadcastHost() or host->isMulticastHost())) {
+      m->numberBroadOrMulti++;
+    } else {
+      if (host->getNumPktsSent()>0 and host->getNumPktsRcvd()>0) {
+        m->numberActive++;
+      } else {
+        m->numberPassive++;
+      }
+    }
+  }
+
+  return(false); //false = keep on walking
+}
+
+/* *************************************** */
+
+int NetworkInterface::countActiveLocalHosts(lua_State* vm, bool treeMapMode) {
+
+  u_int32_t begin_slot = 0;
+  localhost_walker m;
+  int rc;
+
+  m.numberActive= 0;
+  m.numberBroadOrMulti= 0;
+  m.numberPassive= 0;
+
+  rc = walker(&begin_slot, true, walker_hosts, active_localhosts_walker, (void*)&m);
```

Comment:
Can you suggest how to avoid scanning the interface to compute these fields?

Reply:
This problem is still pending, namely, we cannot do a full hash scan whenever a page is rendered. So this work is good as a starting point but it needs to be consolidated before merge